requests/models.py

```diff
@@ -363,8 +364,8 @@ def prepare_url(self, url, params):
 
         # Only want to apply IDNA to the hostname
         try:
-            host = host.encode('idna').decode('utf-8')
-        except UnicodeError:
+            host = idna.encode(host).decode('utf-8')
```

Comment:
Aha, good spot. Yes, ok, that makes some sense. Let's enable the UTS46 mapping then, as there seems to be no reason not to.